web/src/message_list_view.ts

```diff
@@ -216,29 +258,34 @@ function get_users_for_recipient_row(message) {
         };
     });
 
-    function compare_by_name(a, b) {
+    function compare_by_name(a: RecipientRowUser, b: RecipientRowUser): number {
         return util.strcmp(a.full_name, b.full_name);
     }
 
     return users.sort(compare_by_name);
 }
 
-let message_id_to_focus_after_processing_message_events = {
+let message_id_to_focus_after_processing_message_events: {
+    id: number | undefined;
+    selectionStart: number | undefined;
+    selectionEnd: number | undefined;
+} = {
```

Comment:
Would it make sense to type this as all-undefined or all-numbers? Might save you some type assertions later.

Reply:
I don't think it saves type assertions, but it is more accurate so I'll do that.